packages/client/src/containers/Signup/Signup.container.js

```diff
@@ -1,62 +1,103 @@
 import React, { useEffect, useState } from 'react';
-import { Link, useNavigate } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
 import { useUserContext } from '../../userContext';
 import './Signup.styles.css';
+import SignupForm from './SignupForm';
+import validateForm from '../../utils/validateForm';
 
 function Signup() {
   const { user, loading, registerWithEmailAndPassword, signInWithGoogle } =
     useUserContext();
-  const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
-  const [name, setName] = useState('');
+  const [formValues, setFormValues] = useState({
+    fullname: '',
+    email: '',
+    password: '',
+  });
+  const [formErrors, setFormErrors] = useState({});
   const navigate = useNavigate();
-  const register = () => {
-    /* if (!name) alert('Please enter name'); add some error message & validation */
-    registerWithEmailAndPassword(name, email, password);
+
+  const handleChange = (e) => {
+    setFormValues({ ...formValues, [e.target.name]: e.target.value });
+  };
+
+  const handleValidation = () => {
+    setFormErrors(validateForm(formValues));
+  };
+
+  const cleanUpValidation = () => {
+    setFormErrors({
+      fullname: '',
+      email: '',
+      password: '',
+    });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (Object.keys(formErrors).length === 0) {
+      registerWithEmailAndPassword(
+        formValues.fullname,
+        formValues.email,
+        formValues.password,
+      );
+    }
```

Comment:
I recommend getting into the habit of flipping your `if` statements, this is called guarding. 
You find the early path to exit, which thanks to the return allows you to keep the rest of the code at the base indent level making it more readable.

```suggestion
    if (Object.keys(formErrors).length !== 0) {
      return;
      // Probably want to do something more here, like inform the user about why the form was not allowed to be submitted.
    }
    
      registerWithEmailAndPassword(
        formValues.fullname,
        formValues.email,
        formValues.password,
      );
    }
```

Reply:
fixed